trade-finance-manager-api/src/v1/controllers/user/user.controller.js

```diff
@@ -28,11 +28,18 @@ exports.findByUsername = async (username, callback) => {
   collection.findOne({ username: { $eq: username } }, { collation: { locale: 'en', strength: 2 } }, callback);
 };
 
-exports.create = async (user, callback) => {
+/**
+ * @param {object} user to create
+ * @param {import('../../../types/tfm-session-user').TfmSessionUser | undefined} sessionUser logged in user
+ * @param {(error: string | null, createdUser: object) => void} callback
+ * @returns 
+ */
+exports.create = async (user, sessionUser, callback) => {
   const collection = await db.getCollection('tfm-users');
   const tfmUser = {
     ...user,
     status: USER.STATUS.ACTIVE,
+    auditDetails: sessionUser?._id ? generateTfmUserAuditDetails(sessionUser._id) : null,
```

Comment:
If no user is logged in I've set this to null - not entirely happy this is the best option

Reply:
Can you be not logged in for this one? :( that strikes me as odd is it like that for tests or is it just api key only or smt?

Reply:
I read the comment lower and I'm with you, let's leave like this for now because I don't see anyone actually hitting the not logged in endpoint in production etc.

Reply:
Yeah it's a weird thing - I'll add a comment explaining that the audit logs will be null in development only